src/TLuaInterpreter.cpp

```diff
@@ -6801,30 +6844,19 @@ int TLuaInterpreter::exists( lua_State * L )
     QString name = _name.c_str();
     if( type == "timer")
     {
-        QMap<QString, TTimer *>::const_iterator it1 = pHost->getTimerUnit()->mLookupTable.find( name );
-        while( it1 != pHost->getTimerUnit()->mLookupTable.end() && it1.key() == name )
-        {
-            cnt++;
-            it1++;
-        }
+        cnt += pHost->getTimerUnit()->mLookupTable.count( name );
     }
     else if( type == "trigger")
     {
-        QMap<QString, TTrigger *>::const_iterator it1 = pHost->getTriggerUnit()->mLookupTable.find( name );
-        while( it1 != pHost->getTriggerUnit()->mLookupTable.end() && it1.key() == name )
-        {
-            cnt++;
-            it1++;
-        }
+        cnt += pHost->getTriggerUnit()->mLookupTable.count( name );
     }
     else if( type == "alias")
     {
-        QMap<QString, TAlias *>::const_iterator it1 = pHost->getAliasUnit()->mLookupTable.find( name );
-        while( it1 != pHost->getAliasUnit()->mLookupTable.end() && it1.key() == name )
-        {
-            cnt++;
-            it1++;
-        }
+        cnt += pHost->getAliasUnit()->mLookupTable.count( name );
+    }
+    else if( type == "keybind")
+    {
+        cnt += pHost->getKeyUnit()->mLookupTable.count( name );
```

Comment:
I really think that having a name that implies a boolean value returning an integer is wrong.
```
if (exists("someName", "someType")) {
  //This block will fire on any non-zero return value
}

if (exists("someName", "someType") == true) {
  //This block will only fire on a return value of 1
}
```
It is redundant to use equality to test for a boolean, but I'm oversimplifying. A boolean conditional test will pass on any non-zero integer value, but because `true` evaluates to 1, testing for equality to `true` can fail for reasons that would not be clear to someone who didn't know that nuance.

I had another assumption about collections and exists() functions, but those don't appear to hold up in c++ style. However, it appears that throughout all the Q-Collections, the function exists() doesn't occur. The only function that is implemented is count().

Also, to muddy things more, it looks like lua considers 0 as true in conditionals, which sounds crazy to me, but comes from the Lua.org page on booleans. So even from the lua scripting side, treating the exists() call like a boolean will fail in ways that are inconsistent with other languages I'm familiar with.

So I guess what I would argue is that you're right that functionality shouldn't be changed, but I think the name should be refactored for clarity and consistency. Feel free to disregard this, it's your project, and I understand not wanting to break api like that, I just see the potential for (albeit brief) confusion.